### ISSUE TITLE
[NSRegularExpression enumerateMatchesInString:options:range:usingBlock:]: Range or index out of bounds

### ISSUE THREAD

**Issue by janyou** [soffes/SyntaxKit#9]
use AttributedParser  for .m file

```
@implementation Car

- (int) getOdometerReading
{
    SpecificCar *myCar = [[SpecificCar alloc] init];
    [myCar getOdometerReading];
    return 50000;
}
@end
```

crashed with:
**\* -[NSRegularExpression enumerateMatchesInString:options:range:usingBlock:]: Range or index out of bounds

The crash seems to be happening in Parser.swift 

```
       /// Returns new location
    private func parse(string: String, inRange bounds: NSRange, callback: Callback) -> UInt {
        ...
          let endBounds = NSMakeRange(location, bounds.length - location - bounds.location) 
                  //when endBounds.length = -106  

```

**Comment by rnaud** [comment]
Having the same issue, I believe it's happening here:

```
matches = expression.matchesInString(string, options: [], range: bounds)
```

We are sometimes getting bounds that seem illogical. We have negative bounds.length it seems.

**Comment by soffes** [comment]
Can you post some sample code that reproduces this issue?

**Comment by soffes** [comment]
Fixed in #14
